### ISSUE TITLE
Switch from PhantomJS to wkhtmltopdf

### ISSUE THREAD

**Issue by Web-eWorks** [microluxe20/microluxe20#59]
PhantomJS, as nice as it is, is an absolutely terrible solution for generating PDF documents from markdown sources.

It is slow (sometimes taking upwards of 30 seconds to generate a release), buggy (having non-obvious issues with font and image loading), obtuse (there's no way to debug resource loading or DOM properties like layout), and worst of all, it renders page content as static images and embeds the images into the PDF, instead of generating the appropriate PDF text directives. This means that core features PDF features like search and text selection are impossible - how do you search an image?

Instead, we will be investigating and ultimately switching to either `pandoc` or `wkhtmltopdf`, depending upon ease of use, speed, and developer overhead. This will allow us to have properly searchable and selectable PDF documents, enable support for PDF outlines, and allow us to drop a large, slow dependency.

I am currently leaning towards wkhtmltopdf due to it's lightweight nature and the fact that pandoc depends on 60+ MB of haskell modules, but feature support may change the eventual decision.

As an addendum, this will likely require us to upgrade our travis environment to Ubuntu 16.04 Xenial, as it ships newer versions of the packages we'll require.

**Comment by Web-eWorks** [comment]
Good(-ish) news! `wkhtmltopdf` does pretty much everything I want it to, with one exception. Without having the patched-QT version of the program, fonts are rendered as bezier curves instead of text, bloating the file size and putting us back to square one. (It appears that's what PhantomJS is doing as well, probably because they're both invoking Webkit the same way.)

This really isn't a problem for anyone but me, because the Solus maintainers ship the non-patched-QT version to cut down on dependencies, build time, and size. For debian / ubuntu / windows users (including Travis builds) we should be fine and dandy.

I'll get moving on finalizing the switch soon - we'll be replacing the `gulp-markdown-pdf` package entirely with our own custom `remarkable` + `wkhtmltopdf` process, which also has the added benefit of generating the website's HTML artifacts as a build input for the PDFs.

Now all I've gotta do is track down a unicode formatting bug in our markdown process. Yay.

**Comment by kgrubb** [comment]
I took a minute and tried to play with this branch a bit this evening. There's a lot of linting errors at the moment, but that's probably something thats fairly easy to get up to speed with. This is what I'm currently getting when trying to compile. I included the versions of wkhtmltopdf & node/npm I'm currently using just for reference.
![Screenshot from 2019-05-12 00-55-30](https://user-images.githubusercontent.com/8471701/57577967-b686bd00-7450-11e9-867a-9a104e9ed39a.png)

Looks to be something with the htmlRunner, but not sure what yet.
